src/boot/efi/stub.c

```diff
@@ -180,6 +181,196 @@ static bool use_load_options(
+                }
+
+                err = BS->HandleProtocol(addon,
+                                         MAKE_GUID_PTR(EFI_LOADED_IMAGE_PROTOCOL),
+                                         (void **) &loaded_addon);
+                if (err != EFI_SUCCESS)
+                        return log_error_status(err, "Failed to find protocol in %ls: %m", items[i]);
+
+                err = pe_memory_locate_sections(loaded_addon->ImageBase, unified_sections, addrs, szs);
+                if (err != EFI_SUCCESS || szs[UNIFIED_SECTION_CMDLINE] == 0) {
+                        if (err == EFI_SUCCESS)
+                                err = EFI_NOT_FOUND;
+                        log_error_status(err,
+                                         "Unable to locate embedded .cmdline section in %ls, ignoring: %m",
+                                         items[i]);
+                        continue;
+                }
+
+                /* We want to enforce that addons are not UKIs, i.e.: they must not embed a kernel. */
+                if (szs[UNIFIED_SECTION_LINUX] > 0) {
+                        log_error_status(EFI_INVALID_PARAMETER, "%ls is a UKI, not an addon, ignoring: %m", items[i]);
+                        continue;
+                }
+
+                /* Also enforce that, in case they are specified, .osrel and .uname match */
+                if (osrel && szs[UNIFIED_SECTION_OSREL] > 0 &&
+                                !strneq8(osrel,
+                                         (char *)loaded_addon->ImageBase + addrs[UNIFIED_SECTION_OSREL],
+                                         szs[UNIFIED_SECTION_OSREL])) {
+                        log_error(".osrel mismatch between %ls and UKI, ignoring", items[i]);
+                        continue;
+                }
+                if (uname && szs[UNIFIED_SECTION_UNAME] > 0 &&
+                                !strneq8(uname,
+                                         (char *)loaded_addon->ImageBase + addrs[UNIFIED_SECTION_UNAME],
+                                         szs[UNIFIED_SECTION_UNAME])) {
+                        log_error(".uname mismatch between %ls and UKI, ignoring", items[i]);
+                        continue;
+                }
+
+                _cleanup_free_ char16_t *tmp = TAKE_PTR(buffer),
+                                        *extra16 = xstrn8_to_16((char *)loaded_addon->ImageBase + addrs[UNIFIED_SECTION_CMDLINE],
+                                                                szs[UNIFIED_SECTION_CMDLINE]);
+                buffer = xasprintf("%ls%ls%ls", strempty(tmp), isempty(tmp) ? u"" : u" ", extra16);
+        }
+
+        mangle_stub_cmdline(buffer);
+
+        if (!isempty(buffer)) {
+                _cleanup_free_ char16_t *tmp = TAKE_PTR(*cmdline_append);
+                bool m = false;
+
+                (void) tpm_log_load_options(buffer, &m);
+                *ret_parameters_measured = *ret_parameters_measured < 0 ? m : (*ret_parameters_measured && m);
```

Comment:
please don't. we don't read parameters called `ret_*`, because those are return params, not input/output parameters.

I'd just leave this a return param, and merge this with the ternary op in the caller

Reply:
(simply to remain systematic)